src/main/java/java/lang/String.java

```diff
@@ -1390,11 +1390,33 @@ else if(c<=0xFFFF)
      * @see  #equalsIgnoreCase(String)
      *
      * @diffblue.fullSupport
-     * @diffblue.untested
      */
     public boolean equals(Object anObject) {
-        // DIFFBLUE MODEL LIBRARY This is treated internally in CBMC
-        return CProver.nondetBoolean();
+        // if (this == anObject) {
+        //     return true;
+        // }
+        // if (anObject instanceof String) {
+        //     String anotherString = (String)anObject;
+        //     int n = value.length;
+        //     if (n == anotherString.value.length) {
+        //         char v1[] = value;
+        //         char v2[] = anotherString.value;
+        //         int i = 0;
+        //         while (n-- != 0) {
+        //             if (v1[i] != v2[i])
+        //                 return false;
+        //             i++;
+        //         }
+        //         return true;
+        //     }
+        // }
+        // return false;
+
+        // DIFFBLUE MODEL LIBRARY Use CProverString function
```

Comment:
:pick: In the other PR, it says "method" instead of "function".

Reply:
corrected in the other PR